components/storage/blockdevice/COMPONENT_SPIF/TESTS/block_device/spif/main.cpp

```diff
@@ -47,6 +47,11 @@ void basic_erase_program_read_test(SPIFBlockDevice &block_device, bd_size_t bloc
 {
     int err = 0;
     _mutex->lock();
+
+    // Make sure block address per each test is unique
+    static unsigned block_seed = 1;
+    srand(block_seed++);
```

Comment:
I guess this is sufficient (block seed) for the test purpose or rather using time like getting from us ticker ?

Reply:
I didn't want to make a drastic change. I think this should be sufficient for this test purpose.
We simply need ~50 different addresses for this test, they don't have to be random per each test.I'm saying ~50 ((5 Threads x 10 iterations) because addresses may repeat as long as they do not run concurrently of different threads.

Because seed/rand are not really random, I believe once the test runs successfully it will remain stable. Using time will actually 'random' and then we might have collision issue (very very low probability as David mentioned below)

Reply:
Can you add this to the commit, will help